python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -954,113 +954,117 @@ export default function AIConfigEditor({
-                      >
-                        Save
-                      </Button>
-                    </Tooltip>
-                  )}
-                </Group>
-              }
-            </Flex>
-            <ConfigNameDescription
-              name={aiconfigState.name}
-              description={aiconfigState.description}
-              setDescription={onSetDescription}
-              setName={onSetName}
-            />
-          </div>
-          <GlobalParametersContainer
-            initialValue={aiconfigState?.metadata?.parameters ?? {}}
-            onUpdateParameters={onUpdateGlobalParameters}
-          />
-          <PromptsContainer
-            cancelRunPrompt={callbacks?.cancel}
-            defaultModel={aiconfigState.metadata.default_model}
-            getModels={callbacks?.getModels}
-            onAddPrompt={onAddPrompt}
-            onChangePromptInput={onChangePromptInput}
-            onChangePromptName={onChangePromptName}
-            onDeletePrompt={onDeletePrompt}
-            onRunPrompt={onRunPrompt}
-            onUpdatePromptModel={onUpdatePromptModel}
-            onUpdatePromptModelSettings={onUpdatePromptModelSettings}
-            onUpdatePromptParameters={onUpdatePromptParameters}
-            prompts={aiconfigState.prompts}
-            runningPromptId={runningPromptId}
+                  </Tooltip>
+                )}
+              </Group>
+            }
+          </Flex>
+          <ConfigNameDescription
+            name={aiconfigState.name}
+            description={aiconfigState.description}
+            setDescription={onSetDescription}
+            setName={onSetName}
           />
-        </Container>
-      </AIConfigContext.Provider>
+        </div>
+        <GlobalParametersContainer
+          initialValue={aiconfigState?.metadata?.parameters ?? {}}
+          onUpdateParameters={onUpdateGlobalParameters}
+        />
+        <PromptsContainer
+          cancelRunPrompt={callbacks?.cancel}
+          defaultModel={aiconfigState.metadata.default_model}
+          getModels={callbacks?.getModels}
+          onAddPrompt={onAddPrompt}
+          onChangePromptInput={onChangePromptInput}
+          onChangePromptName={onChangePromptName}
+          onDeletePrompt={onDeletePrompt}
+          onRunPrompt={onRunPrompt}
+          onUpdatePromptModel={onUpdatePromptModel}
+          onUpdatePromptModelSettings={onUpdatePromptModelSettings}
+          onUpdatePromptParameters={onUpdatePromptParameters}
+          prompts={aiconfigState.prompts}
+          runningPromptId={runningPromptId}
+        />
+      </Container>
+    </AIConfigContext.Provider>
+  );
+}
+
+// Wrap the AIConfigEditor in the NotificationProvider to provide NotificationContext
+// to the AIConfigEditor. Wrap NotificationProvider and AIConfigEditor with the
+// theme provider to ensure all components have the proper theme
+export default function AIConfigEditorWrapper(props: Props) {
```

Comment:
Nit: Would prefer to rename this as `AIConfigEditor` so it's more clear this is what is referred to when doing default imports. I get that with default we can call it anything, but for tracing code I think it would be more clear to have this with that name. My proposal would be:
1. Remove `<AIConfigContext.Provider>` tag from current `AIConfigEditor`, put it within current `AIConfigEditorWrapper` component instead
2. Rename `AIConfigEditor` to something like `AIConfigEditorBase` or `AIConfigEditorWithoutContexts` 
3. Rename `AIConfigEditorWrapper` or to `AIConfigEditor`

Reply:
I can rename. I can't move the AIConfigContext.Provider since most of the context values (callbacks) use `showNotification` from context and need to be a child of the notification context provider, which is why the components are separated here

Reply:
Ah ok makes sense, but yea let's rename, even if it's a bit bulky

Reply:
Will do. Long-term, we can move all of the context / callback code in the base component into their own hooks to reduce the size of this file drastically. But holding offf on that for now since it'll be a pain to deal with for the packaging stuff